rust-bins/Cargo.toml

```diff
@@ -13,12 +13,11 @@ vendored-ssl = ["openssl-sys/vendored"]
 [dependencies]
 clap = "2.33"
 dialoguer = "0.10"
-pairing = "0.15"
-rand = "=0.7"
+rand = "=0.8"
 serde = "1.0"
 serde_json = "1.0"
-ed25519-dalek = "=1.0"
-curve25519-dalek = "3.0"
+ed25519-dalek = { version = "2.0", features = ["serde"]}
```

Comment:
`serde` is used here https://github.com/Concordium/concordium-base/blob/arkworks-integration/rust-bins/src/bin/keygen.rs#L309

Reply:
That should be rendered in hex though. Which is different from this instance.

So we should do
```
base16_encode_string(&sig)
```
if the type is the dalek Signature.

Reply:
Is my understanding correct, that in this fragment
```Rust
let sig = keypair.sign(&public_bytes);
    let level_str = match kgup.level {
        Level::Root => "root",
        Level::One => "level1",
        Level::Two => "level2",
    };
let public_data = serde_json::json!({
        "key": {
            "verifyKey": base16_encode_string(&signing_key.verifying_key()),
            "scheme": "Ed25519",
        },
        "signature": sig,
        "type": level_str,
    });
```
I change ` "signature": sig`  to `"signature": base16_encode_string(&sig)`. Then there is no need for a `SerdeSerialize` instance for the signature (since it's a hex string now).
,

Reply:
Fixed as it is described above.